presto-main/src/main/java/io/prestosql/execution/buffer/LazyOutputBuffer.java

```diff
@@ -164,7 +167,7 @@ public void setOutputBuffers(OutputBuffers newOutputBuffers)
                         delegate = new PartitionedOutputBuffer(taskInstanceId, state, newOutputBuffers, maxBufferSize, systemMemoryContextSupplier, executor);
                         break;
                     case BROADCAST:
-                        delegate = new BroadcastOutputBuffer(taskInstanceId, state, maxBufferSize, systemMemoryContextSupplier, executor, notifyStatusChanged);
+                        delegate = new BroadcastOutputBuffer(taskInstanceId, state, maxBroadcastBufferSize, systemMemoryContextSupplier, executor, notifyStatusChanged);
```

Comment:
is it better that we have it as `max(sink.max-buffer-size, sink-.max-broadcast-buffer-size)`? and mention that in the documentation? so that if someone wants to bump the generic buffer size, the broadcast buffer also gets bumped up automatically.

Reply:
I don't think it's needed to use `max` between the two. The reason is that amount of data in broadcast buffer should already be capped with CBO (e.g why to increase it above 200MB if CBO broadcast limit is still 100MB?). Additionally, using `max` would add some extra logic here (instead of simple toggle).

> mention that in the documentation?

documentation will be in follow-up PR.

Reply:
> I don't think it's needed to use max between the two. The reason is that amount of data in broadcast buffer should already be capped with CBO (e.g why to increase it above 200MB if CBO broadcast limit is still 100MB?). Additionally, using max would add some extra logic here (instead of simple toggle).

Yes, that's right for the default case with CBO. What if someone is tuning the `sink.max-buffer-size` config? Ideally wouldn't we want the broadcast buffer to be as much as normal buffer? (probably by just having an assertion that the max-broadcast-buffer >= max-buffer, but that could break the server start with existing configs). I am ok if you think that the documentation can cover these things.

Reply:
> Ideally wouldn't we want the broadcast buffer to be atleast as much as normal buffer?

In case of non-broadcast buffer it is important to ensure seamless flow of data without buffer trashing. Non-broadcast exchanges can transmit TBs of data.
In case of broadcast buffer it's more about storage of data so that source tasks can start flushing. Broadcasted data would normally be tiny compared to hash-exchanged data.

> Ideally wouldn't we want the broadcast buffer to be atleast as much as normal buffer?

I would consider 200MB a big buffer already. I doubt 200MB would be that beneficial for hash exchange (considering 32MB is sufficient in most of cases). IMO having 200MB buffer for hash exchange would mostly introduce additional memory overhead (in case probe side is "slow" to fetch data)

Reply:
> Broadcasted data would normally be tiny compared to hash-exchanged data. 
I would consider 200MB a big buffer already. I doubt 200MB would be that beneficial for hash exchange (considering 32MB is sufficient in most of cases).

Yes, that's true.